surface/include/pcl/surface/marching_cubes.h

```diff
@@ -355,6 +355,8 @@ namespace pcl
     *
     * Lorensen W.E., Cline H.E., "Marching cubes: A high resolution 3d surface construction algorithm",
     * SIGGRAPH '87
+    * 
+    * \tparam PointNT A point type similar to pcl::PointNormal
```

Comment:
This is a bit unclear IMO: what exactly does "similar" mean? Is `pcl::Normal` similar? Is `PointXYZLNormal` similar?
Looking at [the .cpp file](https://github.com/PointCloudLibrary/pcl/blob/master/surface/src/marching_cubes.cpp#L42), only `pcl::PointNormal`, `pcl::PointXYZRGBNormal`, and `pcl::PointXYZINormal` are instantiated (although on a quick glance, no normal information is used in `MarchingCubes`, so maybe it would even work with other point types?).
I remember that this confusion about which point types are instantiated happened [for other classes too](https://github.com/PointCloudLibrary/pcl/issues/4998). Often, it seems arbitrary which point types are instantiated, especially when `PCL_ONLY_CORE_POINT_TYPES` is defined.
[This tutorial](https://pcl.readthedocs.io/projects/tutorials/en/master/adding_custom_ptype.html) explains the instantiations, but given that these questions arise from time to time, it might not be visible enough?
Adding to the confusion, `PointNT` seems to roughly mean "point type with xyz and normal information" here, but elsewhere it sometimes only means "point type with normal information".
Honestly not sure what the best solution to all these questions is. For now, we could simply list the three possible point types for `PointNT` here?

Reply:
TBH, I prefer compiler aided stuff like `HasNormal<T>` and family. When I have time, I'd take a crack at that with MarchingCubes